app/src/main/java/com/example/cinemapp/ui/main/ProfileFragment.kt

```diff
@@ -1,65 +1,184 @@
 package com.example.cinemapp.ui.main
 
 import android.content.Intent
-import androidx.fragment.app.viewModels
 import android.os.Bundle
 import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.core.content.ContextCompat
 import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.LinearSmoothScroller
+import androidx.recyclerview.widget.RecyclerView
 import com.example.cinemapp.R
-import com.example.cinemapp.data.UserPreferences
 import com.example.cinemapp.databinding.FragmentProfileBinding
-import com.example.cinemapp.databinding.FragmentSearchBinding
 import com.example.cinemapp.ui.authentication.AuthenticationActivity
+import com.example.cinemapp.ui.main.model.AccountDetails
+import com.example.cinemapp.ui.main.model.MovieCard
+import com.example.cinemapp.util.isEndOfScroll
+import com.example.cinemapp.util.loadImage
+import com.example.cinemapp.util.observeFlowSafely
+import com.example.cinemapp.util.safeNavigateWithArgs
+import com.google.android.material.tabs.TabLayout
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
-import org.koin.android.ext.android.inject
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class ProfileFragment : Fragment() {
 
     private val viewModel by viewModel<ProfileViewModel>()
+    private val adapter = MovieAdapter()
     private var _binding: FragmentProfileBinding? = null
     private val binding get() = _binding!!
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-
-    }
-
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
         _binding = FragmentProfileBinding.inflate(inflater, container, false)
+        setupVisibility()
+        setupAdapter()
+        viewModel.getAccountDetails()
```

Comment:
Here, fragment has some knowledge of what data to collect. You might want to encapsulate fetching logic inside view model, and call it something like `loadData`